code/game/objects/items/weapons/tools/knives.dm

```diff
@@ -165,4 +165,106 @@
 	if(prob(min(100,(100-L.getarmor(user.targeted_organ, ARMOR_MELEE))+modifier)))
 		var/trans = reagents.trans_to_mob(target, rand(1,3)*reagent_modifier, CHEM_BLOOD)
 		admin_inject_log(user, target, src, reagents.log_list(), trans)
-		to_chat(user, SPAN_NOTICE("You inject [trans] units of the solution. [src] now contains [src.reagents.total_volume] units."))
\ No newline at end of file
+		to_chat(user, SPAN_NOTICE("You inject [trans] units of the solution. [src] now contains [src.reagents.total_volume] units."))
+
+/obj/item/weapon/tool/knife/butterfly
+	name = "butterfly knife"
+	desc = "A basic metal blade concealed in a lightweight plasteel grip. Small enough when folded to fit in a pocket."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "butterflyknife"
+	item_state = "butterflyknife"
+	flags = CONDUCT
+	edge = FALSE
+	sharp = FALSE
+	force = WEAPON_FORCE_WEAK
+	switched_on_force = WEAPON_FORCE_PAINFUL
+	matter = list(MATERIAL_PLASTEEL = 4, MATERIAL_STEEL =6)
+	switched_on_qualities = list(QUALITY_CUTTING = 20, QUALITY_WIRE_CUTTING = 10, QUALITY_SCREW_DRIVING = 5)
+	tool_qualities = null
+	toggleable = TRUE
+
+/obj/item/weapon/tool/knife/butterfly/turn_on(mob/user) 
+	item_state = "[initial(item_state)]_on"
+	to_chat(user, SPAN_NOTICE("You flip out [src]."))
+	playsound(user, 'sound/weapons/flipblade.ogg', 15, 1)
+	edge = TRUE
+	sharp = TRUE
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
+	switched_on = TRUE
+	tool_qualities = switched_on_qualities
+	if (!isnull(switched_on_force))
+		force = switched_on_force
+	if(glow_color)
+		set_light(l_range = 1.7, l_power = 1.3, l_color = glow_color)
+	START_PROCESSING(SSobj, src)
```

Comment:
What is this processing?

Reply:
leftover from the proc I copied over, im guessing to activate glowing in an actual powered tool. Gonna remove this since switchblades are unpowered and shouldnt glow.